### PR TITLE
Issue #6207: Add Xpath Regression Test for FinalParameters

### PR DESCRIPTION
**Pull request by Zopsss**
Partial for issue #6207 Add xpath regression test for FinalParameters.

Link to the documentation: https://checkstyle.sourceforge.io/checks/misc/finalparameters.html

### CODE REVIEW

**Comment by MauryanKansara** [comment]
I'm facing an error while running `mvn clean verify`, here is the error log:

```
[ERROR] Errors: 
[ERROR]   PropertyCacheFileTest.testMultipleSymbolicLinkResolution:350 ┬╗ FileSystem C:\Users\maury\AppData\Local\Temp\firstLinkDir16234361358164004\firstLink: A required privilege is not held by the client
[ERROR]   PropertyCacheFileTest.testPersistWithSymbolicLinkToDirectory:289 ┬╗ FileSystem C:\Users\maury\AppData\Local\Temp\symbolicLinkDir15327163968903169029\symbolicLink: A required privilege is not held by the client
[ERROR]   PropertyCacheFileTest.testSymbolicLinkResolution:308 ┬╗ FileSystem C:\Users\maury\AppData\Local\Temp\symbolicLinkDir12026308835976699930\symbolicLink: A required privilege is not held by the client
[ERROR]   PropertyCacheFileTest.testSymbolicLinkToNonDirectory:328 ┬╗ FileSystem C:\Users\maury\AppData\Local\Temp\symbolicLinkDir1109373640951616353\symbolicLink: A required privilege is not held by the client
```

can anyone help me out?

**Comment by rnveach** [comment]
@MauryanKansara See https://github.com/checkstyle/checkstyle/issues/14219

**Comment by romani** [comment]
@MauryanKansara, please review other CI failures and address them.

**Comment by romani** [comment]
@MauryanKansara , please consider workaround https://github.com/checkstyle/checkstyle/issues/14219#issuecomment-1876387122

**Comment by MauryanKansara** [comment]
@romani should I add more example code to the input files or the current input files are ok?

File: src/it/resources/org/checkstyle/suppressionxpathfilter/finalparameters/SuppressionXpathRegressionFinalParameters3.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.checkstyle.suppressionxpathfilter.finalparameters;
+
+public class SuppressionXpathRegressionFinalParameters3 {
+
+    public void method(String[] arg) { // warn
```
Comment (romani) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/finalparameters/SuppressionXpathRegressionFinalParameters3.java:5]: This is kind of the same as SuppressionXpathRegressionFinalParameters1

Please update this file to report violation on method of anonymous class

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/it/resources/org/checkstyle/suppressionxpathfilter/finalparameters/SuppressionXpathRegressionFinalParameters3.java (hunk 1 above)
Comment (MauryanKansara) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/finalparameters/SuppressionXpathRegressionFinalParameters3.java]: @romani done

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
Thanks a lot

**Pull request merged**